src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java

```diff
@@ -186,14 +187,11 @@ private Card createJobCompletedCard() {
                 summary += " Back to Normal";
 
                 if (failingSinceRun != null) {
-                    long duration = run.getDuration() == 0L
-                            ? System.currentTimeMillis() - run.getStartTimeInMillis()
-                            : run.getDuration();
-                    long currentBuildCompletionTime = run.getStartTimeInMillis() + duration;
+                    long currentBuildCompletionTime = TimeUtils.countCompletionTime(run.getStartTimeInMillis(), run.getDuration());
                     factsBuilder.addBackToNormalTime(currentBuildCompletionTime - failingSinceRun.getStartTimeInMillis());
                 }
-            } else if (result == Result.FAILURE && failingSinceRun != null) {
-                if (previousResult == Result.FAILURE) {
+            } else if (result == Result.FAILURE) {
```

Comment:
when result was FAILURE but failingSinceRun was null then the last case of this if was reached and the status was FAILURE not Failure